3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/AnonIpEnrichment.scala

```diff
@@ -118,23 +167,60 @@ case class AnonIpEnrichment(
    * anonymizeIp("94.15.223.151", Three)
    * => "94.x.x.x"
    *
+   *
+   * For IPv6 either the form defined in RFC 2732
+   *  or the literal IPv6 address format defined in RFC 2373 is accepted
+   *
    * TODO: potentially update this to return
    * a Validation error or a null if the IP
    * address is somehow invalid or incomplete.
    *
    * @param ip The IP address to anonymize
    * @return the anonymized IP address
    */
-  import AnonOctets._
-  def anonymizeIp(ip: String): String =
-    Option(ip)
-      .map(
-        _.split("\\.").zipWithIndex
-          .map {
-            case (q, i) => {
-              if (octets.id >= All.id - i) "x" else q
-            }
-          }
-          .mkString("."))
-      .orNull
+  val IPv4MappedAddressPrefix = "::FFFF:"
+  val MaskChar                = "x"
+
+  def anonymizeIp(ipOrNull: String): String =
+    Option(ipOrNull).map { ip =>
+      Try(InetAddress.getByName(ip))
+        .map {
+          case _: Inet4Address => anonymizeIpV4(ip)
+          case _: Inet6Address => anonymizeIpV6(ip)
+        }
+        .getOrElse(tryAnonymizingInvalidIp(ip))
+    }.orNull
+
+  private def anonymizeIpV4(ip: String): String = {
+    def masked(ipv4: String) = {
+      val splitted = ipv4.split("\\.")
```

Comment:
```suggestion
      val split = ipv4.split("\\.")
```

Reply:
the `val` contains already splitted ipv4, that's why it's splitted, not split.

Reply:
Yeah, I got that. But `split` is one of those weird words that are the same in the past tense, like `put` and `cut`.

Reply:
🤦‍♂ thanks!